### ISSUE TITLE
Remove development postgresql same-user requirement.

### ISSUE THREAD

**Issue by ruckc** [oneis/haplo#5]
The development postgresql same-user requirement is foolish.  Please look into removing this development requirement.  This complicates development environment setup, as most databases for development still run as separate users.  To my knowledge there is no good reason that can't be handled via postgresql user permissions and pg_hba.conf, to require same-user.

**Comment by bensummers** [comment]
Yes, it's not ideal. However, it speeds up the tests by about 10 times, making them far less painful to run.

We compile and build a pg extension for managing Xapian databases within the pg process. Tests need to be able to snapshot and restore the Xapian databases to get into a known state to begin a test, so the test process needs to have read/write access.

Now of course it could be done just as well by messing around with permissions, but this way matches our internal development environment, which is set up completely with Puppet and scripts.

I'm not going to do anything about this right now, but I am going to leave this issue open for when we do the final version of the build script. When I running through the build instructions to check them, I did find postgresql part a bit painful when it wasn't automated away by a bunch of scripts.

Thank you!

**Comment by DRMacIver** [comment]
Suggested fix:

Given the benefit you derive from having same user postgres, it seems obviously sensible to keep it. The only annoying thing is having to manually set this up yourself as a user. In order to avoid doing this you could have the tests explicitly manage launching and shutting down a same-user postgres install, and run it on a non-standard port so that you don't have to worry about it conflicting with an existing postgres install. Does this seem reasonable?

**Comment by bensummers** [comment]
That's a pretty good idea. It does require that the fetch-and-build.sh script store knowledge of where the database server is internally, but given the other messing around with postgres there I don't think it's a major deal.

Also needs to do this for script/server where you're trying the user interface in development mode.

(Here we have a scripted install of the database as the developer's user, so we don't see any of this hassle. The advantages of a very constrained environment.)
